connector-sdk/core/src/main/java/io/camunda/connector/api/error/retry/RetryContext.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.api.error.retry;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class RetryContext {
```

Comment:
The https://docs.camunda.io/docs/8.3/apis-tools/grpc/#output-activatejobsresponse contains the number of retries that are left. Do you see the need to have a dedicated variable for this?

Reply:
Yes we need a new variable, otherwise that would:
- not allow us to use the exponential backoff retry (we need the number of attempts already done)
- not allow to handle different retries (say an error witry retry=2, another with retry=4). We don't know _who_ set the retry value initially.
- some other edge cases that I tried to highlight in UTs/Javadocs

Reply:
I would leave this up to the developer for now.